Mathlib/Algebra/Lie/Weights/Basic.lean

```diff
@@ -184,6 +185,48 @@ lemma weightSpace_le_weightSpaceOf (x : L) (χ : L → R) :
     weightSpace M χ ≤ weightSpaceOf M (χ x) x :=
   iInf_le _ x
 
+variable (R L) in
+/-- A weight of a Lie module is a map `L → R` such that the corresponding weight space is is
+non-trivial. -/
+structure Weight where
+  /-- The family of eigenvalues corresponding to a weight. -/
+  toFun : L → R
+  weightSpace_ne_bot : weightSpace M toFun ≠ ⊥
+
+namespace Weight
+
+instance instFunLike :
+    FunLike (Weight R L M) L R where
+  coe χ := χ.1
+  coe_injective' χ₁ χ₂ h := by cases χ₁; cases χ₂; simp_all
+
+@[simp] lemma coe_weight_mk (χ : L → R) (h) :
+    (↑(⟨χ, h⟩ : Weight R L M) : L → R) = χ :=
+  rfl
+
+variable {M}
+
+/-- The proposition that a weight of a Lie module is zero.
+
+We make this definition because we cannot define a `Zero (Weight R L M)` instance since the weight
+space of the zero function can be trivial. -/
+def IsZero (χ : Weight R L M) := (χ : L → R) = 0
+
+@[simp] lemma coe_eq_zero_iff (χ : Weight R L M) : (χ : L → R) = 0 ↔ χ.IsZero := Iff.rfl
```

Comment:
Is this a good simp lemma for the global simp set? I can imagine both sides are useful, and the RHS isn't clearly "simpler".

Reply:
I'm pretty keen on this being `simp` to establish the normal form as I've built all the API around the `IsZero` definition.